src/Symfony/Component/Notifier/Bridge/LightSms/LightSmsTransport.php

```diff
@@ -0,0 +1,151 @@
+        '2' => 'Login not specified',
+        '3' => 'Text not specified',
+        '4' => 'Phone number not specified',
+        '5' => 'Sender not specified',
+        '6' => 'Invalid signature',
+        '7' => 'Invalid login',
+        '8' => 'Invalid sender name',
+        '9' => 'Sender name not registered',
+        '10' => 'Sender name not approved',
+        '11' => 'There are forbidden words in the text',
+        '12' => 'Error in SMS sending',
+        '13' => 'Phone number is in the blackist. SMS sending to this number is forbidden.',
+        '14' => 'There are more than 50 numbers in the request',
+        '15' => 'List not specified',
+        '16' => 'Invalid phone number',
+        '17' => 'SMS ID not specified',
+        '18' => 'Status not obtained',
+        '19' => 'Empty response',
+        '20' => 'The number already exists',
+        '21' => 'No name',
+        '22' => 'Template already exists',
+        '23' => 'Missing Month (Format: YYYY-MM)',
+        '24' => 'Timestamp not specified',
+        '25' => 'Error in access to the list',
+        '26' => 'There are no numbers in the list',
+        '27' => 'No valid numbers',
+        '28' => 'Missing start date (Format: YYYY-MM-DD)',
+        '29' => 'Missing end date (Format: YYYY-MM-DD)',
+        '30' => 'No date (format: YYYY-MM-DD)',
+        '31' => 'Closing direction to the user',
+        '32' => 'Not enough money',
+        '33' => 'Missing phone number',
+        '34' => 'Phone is in stop list',
+        '35' => 'Not enough money',
+        '36' => 'Can not obtain information about phone',
+        '37' => 'Base Id is not set',
+        '38' => 'Phone number already exists in this database',
+        '39' => 'Phone number does not exist in this database',
+    ];
+
+    public function __construct(string $login, string $password, string $phone, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $login;
+        $this->password = $password;
+        $this->phone = $phone;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('lightsms://%s?phone=%s', $this->getEndpoint(), $this->phone);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $signature = $this->generateSignature($message);
+
+        $endpoint = sprintf(
+            'https://%s/external/get/send.php?login=%s&signature=%s&phone=%s&text=%s&sender=%s&timestamp=%s',
+            $this->getEndpoint(),
+            $this->login,
+            $signature,
+            str_replace('+', '', $message->getPhone()),
```

Comment:
Yes. Provider do not allow to send with plus

Reply:
Yeah, I understand that, but removing the `+` does not make the phone number valid.
Usually the international prefix is `00`.

Side note: given the parameters were not escaped, I wonder if the request failed because the provider does not accept the `+` or because the char has been converted by a space...

Reply:
In my case with + in Lithuania I got this one
```
Unable to send the SMS: Invalid signature
```

Reply:
Lithuanian SMS provider down, but I can use
00370 or 370, but if I use +370 than I got error about signature.

Reply:
> Unable to send the SMS: Invalid signature

Make me think, the values used to compute the signature is different from the values send in the request, which make me think the issue is about parameters not escaped.

Anyway, if `+` is forbidden, I'd rather replace it by `00` instead.

Reply:
You think better to use option 2?

```
(1) preg_replace("/[^\d]/", '', $phoneNumber);
(2) str_replace('+', '00', $phoneNumber);
```

Reply:
(2) - changed